pkg/types/azure/validation/platform.go

```diff
@@ -68,18 +69,44 @@ func ValidatePlatform(p *azure.Platform, publish types.PublishingStrategy, fldPa
 	if p.DefaultMachinePlatform != nil {
 		allErrs = append(allErrs, ValidateMachinePool(p.DefaultMachinePlatform, "", p, nil, fldPath.Child("defaultMachinePlatform"))...)
 	}
-	if p.VirtualNetwork != "" {
-		if p.ComputeSubnet == "" {
-			allErrs = append(allErrs, field.Required(fldPath.Child("computeSubnet"), "must provide a compute subnet when a virtual network is specified"))
+	hasControlPlane := false
+	hasCompute := false
+	subnetSpecList := map[string]bool{}
+	for _, subnets := range p.Subnets {
```

Comment:
```suggestion
	for _, subnet := range p.Subnets {
```

nit: we should singular here as it holds specs for each subnet.